client/post-editor/media-modal/index.jsx

```diff
@@ -127,6 +127,7 @@ export class EditorMediaModal extends Component {
 		return {
 			filter: '',
 			detailSelectedIndex: 0,
+			source: '',
```

Comment:
This is the root for `source` and `onSourceChange`, matching `search` and `filter` - there's nothing passed in.